test/provider_tracking.cpp

```diff
@@ -0,0 +1,370 @@
+        memory_size = FIXED_BUFFER_SIZE;
+        memory_buffer = malloc(memory_size);
+        ASSERT_NE(memory_buffer, nullptr);
+
+        // Create provider parameters
+        umf_fixed_memory_provider_params_handle_t params = nullptr;
+        umf_result_t res = umfFixedMemoryProviderParamsCreate(
+            &params, memory_buffer, memory_size);
+        ASSERT_EQ(res, UMF_RESULT_SUCCESS);
+        ASSERT_NE(params, nullptr);
+
+        providerCreateExt(std::make_tuple(umfFixedMemoryProviderOps(), params),
+                          &provider);
+
+        umfFixedMemoryProviderParamsDestroy(params);
+        umf_result_t umf_result =
+            umfMemoryProviderGetMinPageSize(provider.get(), NULL, &page_size);
+        ASSERT_EQ(umf_result, UMF_RESULT_SUCCESS);
+
+        page_plus_64 = page_size + 64;
+
+        umf_memory_pool_handle_t hPool = nullptr;
+        umf_result = umfPoolCreate(umfProxyPoolOps(), provider.get(), nullptr,
+                                   0, &hPool);
+        ASSERT_EQ(umf_result, UMF_RESULT_SUCCESS);
+
+        pool = umf::pool_unique_handle_t(hPool, &umfPoolDestroy);
+    }
+
+    void TearDown() override {
+        if (memory_buffer) {
+            free(memory_buffer);
+            memory_buffer = nullptr;
+        }
+        test::TearDown();
+    }
+
+    umf::provider_unique_handle_t provider;
+    umf::pool_unique_handle_t pool;
+    size_t page_size;
+    size_t page_plus_64;
+    void *memory_buffer = nullptr;
+    size_t memory_size = 0;
+};
+
+static void
+createPoolFromAllocation(void *ptr0, size_t size1,
+                         umf_memory_provider_handle_t *_providerFromPtr,
+                         umf_memory_pool_handle_t *_poolFromPtr) {
+    umf_result_t umf_result;
+
+    // Create provider parameters
+    umf_fixed_memory_provider_params_handle_t params = nullptr;
+    umf_result = umfFixedMemoryProviderParamsCreate(&params, ptr0, size1);
+    ASSERT_EQ(umf_result, UMF_RESULT_SUCCESS);
+    ASSERT_NE(params, nullptr);
+
+    umf_memory_provider_handle_t provider1 = nullptr;
+    umf_result = umfMemoryProviderCreate(umfFixedMemoryProviderOps(), params,
+                                         &provider1);
+    ASSERT_EQ(umf_result, UMF_RESULT_SUCCESS);
+    ASSERT_NE(provider1, nullptr);
+
+    umf_memory_pool_handle_t pool1 = nullptr;
+    umf_result =
+        umfPoolCreate(umfProxyPoolOps(), provider1, nullptr, 0, &pool1);
+    ASSERT_EQ(umf_result, UMF_RESULT_SUCCESS);
+
+    umfFixedMemoryProviderParamsDestroy(params);
+
+    *_providerFromPtr = provider1;
+    *_poolFromPtr = pool1;
+}
+
+// TESTS
+
+INSTANTIATE_TEST_SUITE_P(trackingProviderTest, TrackingProviderTest,
+                         ::testing::Values(providerCreateExtParams{
+                             umfFixedMemoryProviderOps(), nullptr}));
+
+TEST_P(TrackingProviderTest, create_destroy) {
+    // Creation and destruction are handled in SetUp and TearDown
+}
+
+TEST_P(TrackingProviderTest, whole_size_success) {
```

Comment:
Do we have any multithreaded tests for the tracking provider itself?

Reply:
AFAIK no, it is tested in the multithreaded tests of memory pools.

Reply:
It would be nice to have a few, but I guess this can be done in a separate PR.

Reply:
I agree that it should be a separate PR. Today tracking provider is indirectly tested in many concurrent scenarios like @ldorau mentioned. There are also IPC concurrent tests that are also tested memory tracker and IPC tracker.